src/segments/umbraco.go

```diff
@@ -0,0 +1,231 @@
+package segments
+
+import (
+	"encoding/xml"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+)
+
+type Umbraco struct {
+	props properties.Properties
+	env   platform.Environment
+
+	FoundUmbraco    bool
+	IsModernUmbraco bool
+	IsLegacyUmbraco bool
+	Version         string
+}
+
+const (
+	umbracoFolderName = "umbraco"
+	umbracoWebConfig  = "web.config"
+)
+
+type CSProj struct {
+	PackageReferences []struct {
+		Name    string `xml:"Include,attr"`
+		Version string `xml:"Version,attr"`
+	} `xml:"ItemGroup>PackageReference"`
+}
+
+type WebConfig struct {
+	AppSettings []struct {
+		Key   string `xml:"key,attr"` // TODO: What happens if the web.config has the attribute as uppercase Key="" ?
+		Value string `xml:"value,attr"`
+	} `xml:"appSettings>add"`
+}
+
+type FindUmbracoResult struct {
+	FoundUmbracoFolder bool
+	FoundWebConfig     bool
```

Comment:
I would make this a ENUM as you only handle one type at a time, so you can switch it.